POSIX/posix-cflags.sh

```diff
@@ -14,6 +14,11 @@ if ! ${CC} -D_POSIX_C_SOURCE=200809L posix-clock_realtime.c 2>/dev/null; then
 	printf %s "-DPOSIXFAIL_CLOCK_REALTIME"
 	echo "WARNING: POSIX violation: <time.h> not defining CLOCK_REALTIME" 1>&2
 fi
+if ! ${CC} -D_POSIX_C_SOURCE=200809L posix-clock_gettime.c 2>/dev/null; then
```

Comment:
Hmm, isn't this orthogonal to the problem it's supposed to be handling?  This detects if `clock_gettime` is *declared*, but I thought the problem was systems where it wasn't being *exposed in libraries* (and the failure only occurring at run time).

Reply:
Yes, the initial report was about a system which declared `clock_gettime()` but did not provide it for linking.  However, that's after a layer of build system + fixes.

On my native 10.11 machine, `clock_gettime()` is *not* declared:
https://github.com/Tarsnap/tarsnap/issues/286#issuecomment-324072189
and this PR fixes that problem.

The related PR that imports this to Tarsnap (https://github.com/Tarsnap/tarsnap/pull/287) also addresses the case of exposing the function but not providing it for linking:
https://github.com/Tarsnap/tarsnap/pull/287/commits/0af569e18e7cf90562106cb90cd0250fbb29ea8e#diff-67e997bcfdac55191033d57a16d1408aR502
via the `$ac_cv_lib_rt_clock_gettime` variable.

So the only case that's missing is POSIX build with libcperciva monoclock.  I admit that I haven't tried compiling spiped with brew yet, but I'm not certain it needs an extra "function is includable but not linkable" fix because the weirdness that got the system into that situation might only apply to brew + autotools.  I'll test this the next time I boot into OSX.

Reply:
Ok, but your patch doesn't change the behaviour on the 10.11 machine you mention, because CLOCK_REALTIME isn't defined either.  You would only see a change in behaviour if CLOCK_REALTIME was defined in `<time.h>` but clock_gettime wasn't, and AFAIK nobody has ever made a system quite *that* broken.